iroha/src/telemetry/dev.rs

```diff
@@ -43,8 +45,8 @@ pub fn start(
     // So let synchonous synchronous code be here.
     //
     // TODO: After migration to tokio move to https://docs.rs/tokio-serde
-    let join_handle = task::spawn_blocking(move || {
-        while let Some(telemetry) = task::block_on(telemetry.next()) {
+    let join_handle = task::spawn(async move {
+        while let Some(telemetry) = telemetry.next().await {
             if let Err(error) = serde_json::to_writer(&mut writer, &telemetry) {
```

Comment:
This write would be blocking. It could lock thread while writing on disk. That is why it was synchronous function beforehand

Reply:
Could you spawn_blocking `serde_json::to_writer`?